cicecore/cicedynB/analysis/ice_diagnostics.F90

```diff
@@ -2118,20 +2109,17 @@ subroutine diagnostic_abort(istop, jstop, iblk, istep1, stop_label)
       if (icepack_warnings_aborted()) call abort_ice(error_message=subname, &
          file=__FILE__, line=__LINE__)
 
-      this_block = get_block(blocks_ice(iblk),iblk)         
-
-      write (nu_diag,*) 'istep1, my_task, iblk =', &
-                         istep1, my_task, iblk
-      write (nu_diag,*) 'Global block:', this_block%block_id
-      if (istop > 0 .and. jstop > 0) &
-      write (nu_diag,*) 'Global i and j:', &
-                          this_block%i_glob(istop), &
-                          this_block%j_glob(jstop) 
-      write (nu_diag,*) 'Lat, Lon:', &
-                         TLAT(istop,jstop,iblk)*rad_to_deg, &
-                         TLON(istop,jstop,iblk)*rad_to_deg
-      write (nu_diag,*) 'aice:', &
-                         aice(istop,jstop,iblk)
+      this_block = get_block(blocks_ice(iblk),iblk)
+
+      call flush_fileunit(nu_diag)
+      if (istop > 0 .and. jstop > 0) then
+         call print_state(trim(stop_label),istop,jstop,iblk)
+      else
+         write (nu_diag,*) subname,' istep1, my_task, iblk =', &
+                            istep1, my_task, iblk
+         write (nu_diag,*) subname,' Global block:', this_block%block_id
```

Comment:
In the old code the global block ID was always printed, here it is only printed if `istop` and `jstop` are not both greater than zero, i.e. if we go through the `else` branch.

I think "Global block" should also be printed by `print_state` to keep the same amount of info. Sorry I missed that on my first review.

Reply:
I'm adding it now.  Good idea.